core/cmd/chains_commands.go

```diff
@@ -0,0 +1,225 @@
+package cmd
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/manyminds/api2go/jsonapi"
+	"github.com/pkg/errors"
+	"github.com/urfave/cli"
+	"go.uber.org/multierr"
+
+	"github.com/smartcontractkit/chainlink/core/chains"
+	"github.com/smartcontractkit/chainlink/core/web/presenters"
+)
+
+var chainHeaders = []string{"ID", "Enabled", "Config", "Created", "Updated"}
+
+// chainCommand returns a cli.Command with subcommands for this type of chain.
+// The chainId cli.Flag must be named "id", but may be String or Int.
+func chainCommand[C chains.Config, R presenters.ChainResource[C], P, P2 any](
+	typ string, client ChainClient[C, R, P, P2], chainID cli.Flag) cli.Command {
+	if flagName := chainID.GetName(); flagName != "id" {
+		panic(fmt.Errorf("chainID flag name must be 'id', got: %s", flagName))
+	}
+	lower := strings.ToLower(typ)
+	article := "a"
+	if strings.ContainsAny(lower[:1], "aeiouy") {
+		article = "an"
+	}
+	return cli.Command{
+		Name:  lower,
+		Usage: fmt.Sprintf("Commands for handling %s chains", typ),
+		Subcommands: cli.Commands{
+			{
+				Name:   "create",
+				Usage:  fmt.Sprintf("Create a new %s chain", typ),
+				Action: client.CreateChain,
+				Flags:  []cli.Flag{chainID},
+			},
+			{
+				Name:   "delete",
+				Usage:  fmt.Sprintf("Delete %s %s chain", article, typ),
```

Comment:
You could remove that `article` magic by changing wording: e.g. `"Delete an existing %s chain"`
Just an idea.

Reply:
Yes! Much better!